unused_closure_parameter false positive on lazy var

When a closure is used in this way, `swiftlint` raise a warning about the variable `lbl` not being used.

```swift
    lazy var label: UILabel = { (lbl: UILabel) -> UILabel in
        lbl.backgroundColor = UIColor.red
        return lbl
    }(UILabel())
```

Thanks for the suggestions. I was able to debug the issue and investigate a little bit using `SourceKitten`. Here what I've found. 

This is the test input:
```swift

3.times { number in
	 number
}


{ ( lbl: UILabel) -> UILabel in
	lbl
}(obj)
```
`SwiftLint`  raises a warning on the second closure
> example.swift:7:5: warning: Unused Closure Parameter Violation: Unused parameter "lbl" in a closure should be replaced with _. (unused_closure_parameter)

Here the relevant output from SourceKitten:

```
"key.length" : 30,
"key.name" : "3.times",
"key.bodyoffset" : 10,
"key.kind" : "source.lang.swift.expr.call",
"key.offset" : 1,
"key.namelength" : 7

"key.length" : 43,
"key.name" : "{ ( lbl: UILabel) -> UILabel in\n\tlbl\n}",
"key.bodyoffset" : 73,
"key.kind" : "source.lang.swift.expr.call",
"key.offset" : 34,
"key.namelength" : 38
``` 

On the first example the closure's body can be found between `key.bodyoffset` and `key.lenghth` and `SwiftLint` code scans that part of the file to find any usage of the parameters.

Unfortunately on the second example the same indexes point outside of the definition of the closure, to the call (`(obj)`) part. Therefor `SwifLint` doesn't find any usage of the parameter and raise the (wrong) warning.

The first solution coming to my mind is to check if the `key.name` is actually a closure and in that case search there any usage of the parameter.
It doesn't look really clean so maybe some of you expert can give me an hint?

Thanks!

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -41,6 +41,10 @@
   [Allen Zeng](https://github.com/allen-zeng)
   [#1175](https://github.com/realm/SwiftLint/issues/1175)
 
+* Ignore `unused_closure_parameter` rule on closures that are called inline.  
+  [Marcelo Fabri](https://github.com/marcelofabri)
+  [#1161](https://github.com/realm/SwiftLint/issues/1161)
+
 ## 0.16.1: Commutative Fabric Sheets
 
 ##### Breaking

--- a/Source/SwiftLintFramework/Rules/UnusedClosureParameterRule.swift
+++ b/Source/SwiftLintFramework/Rules/UnusedClosureParameterRule.swift
@@ -33,7 +33,11 @@ public struct UnusedClosureParameterRule: ASTRule, ConfigurationProviderRule, Co
             "}",
             "genericsFunc { (a: Type, b) in\n" +
                 "a + b\n" +
-            "}\n"
+            "}\n",
+            "var label: UILabel = { (lbl: UILabel) -> UILabel in\n" +
+            "   lbl.backgroundColor = .red\n" +
+            "   return lbl\n" +
+            "}(UILabel())\n"
         ],
         triggeringExamples: [
             "[1, 2].map { ↓number in\n return 3\n}\n",
@@ -81,6 +85,7 @@ public struct UnusedClosureParameterRule: ASTRule, ConfigurationProviderRule, Co
     private func violationRanges(in file: File, dictionary: [String: SourceKitRepresentable],
                                  kind: SwiftExpressionKind) -> [(range: NSRange, name: String)] {
         guard kind == .call,
+            !isClosure(dictionary: dictionary),
             let offset = dictionary.offset,
             let length = dictionary.length,
             let nameOffset = dictionary.nameOffset,
@@ -131,6 +136,14 @@ public struct UnusedClosureParameterRule: ASTRule, ConfigurationProviderRule, Co
             }
             return nil
         }
+    }
+
+    private func isClosure(dictionary: [String: SourceKitRepresentable]) -> Bool {
+        return dictionary.name.flatMap { name -> Bool in
+            let length = name.bridge().length
+            let range = NSRange(location: 0, length: length)
+            return regex("\\A\\s*\\{").firstMatch(in: name, options: [], range: range) != nil
+        } ?? false
     }
 
     private func violationRanges(in file: File,

